pkg/devfile/adapters/kubernetes/component/podwatcher_test.go

```diff
@@ -0,0 +1,301 @@
+package component
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	adaptersCommon "github.com/openshift/odo/pkg/devfile/adapters/common"
+	devfileParser "github.com/openshift/odo/pkg/devfile/parser"
+	"github.com/openshift/odo/pkg/kclient"
+	"github.com/openshift/odo/pkg/machineoutput"
+	"github.com/openshift/odo/pkg/testingutil"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestStatusReconciler(t *testing.T) {
```

Comment:
Should we add test cases to cover the following scenarios?
1) Different uids in pre and post pods? (One scenario I guess this could cover is deployment restart where a new pod gets created and there's multiple pods briefly?)
2) Any scenarios with multiple pods (pre or post)

Reply:
Done.